src/RevTracer.cc

```diff
@@ -182,6 +182,25 @@ void RevTracer::memRead(uint64_t adr, size_t len, void *data)
     traceRecs.emplace_back(TraceRec_t(MemLoad,adr,len,d));
 }
 
+void SST::RevCPU::RevTracer::memhSendRead(uint64_t adr, size_t len, uint16_t reg)
+{
+    traceRecs.emplace_back(TraceRec_t(MemhSendLoad, adr, len, reg));
+}
+
+void RevTracer::memReadResponse(size_t len, void *data, const MemReq* req)
+{
+    if (req->DestReg==0) return;
+    CompletionRec_t c;
```

Comment:
You might want to put all of the assignments into one constructor call, e.g.
```c++
 CompletionRec_t c{req->Hart, req->DestReg, len, req->Addr, data, req->RegType};
 completionRecs.emplace_back(c);
```
and then
```c++
CompletionRec_t(unsigned hart, uint32_t destReg, size_t len, uint64_t addr, void* data, RevRegClass regClass) 
: hart(hart), destReg(destReg), len(len), addr(addr), isFloat(regClass == RevRegClass::RegFLOAT), wait4Completion(true) {
  memcpy(&this->data, data, len > sizeof(this->data) ? sizeof(this->data) : len);
}
```